lib/phase/flatten.js

```diff
@@ -67,6 +67,52 @@ function cleanComplexEnvironment(jobs) {
 }
 
 /**
+ * Converts the simplified steps into the most-verbose format
+ *
+ * This is because the user can provide either:
+ *  - "command to execute"
+ *  - { name: command }
+ *
+ * The launcher expects it in a consistent format:
+ *  - { name: "name", command: "command" }
+ * @method convertSteps
+ * @param  {Object}       jobs   List of jobs (name => job)
+ * @return {Object}              Updated list of jobs after up-converting
+ */
+function convertSteps(jobs) {
+    const newJobs = jobs;
+
+    Object.keys(newJobs).forEach((jobName) => {
+        const steps = clone(Hoek.reach(newJobs, `${jobName}.steps`, {
+            default: []
+        }));
+
+        newJobs[jobName].commands = steps.map((step, index) => {
+            let name;
+            let command;
+
+            switch (typeof step) {
+            case 'object':
+                name = Object.keys(step).pop();
```

Comment:
The key value that is here is a user input so you don't know what it is.

This code is trying to translate it into the `{ name: 'foo', command: 'bar' }` schema